### PR TITLE
Add follower replication support

### PR DESCRIPTION
**Pull request by Fullstop000**

### CODE REVIEW

File: src/raftstore/store/fsm/peer.rs [hunk 1]
```diff
@@ -1446,7 +1465,11 @@ impl<'a, T: Transport, C: PdClient> PeerFsmDelegate<'a, T, C> {
         }
         {
             let mut meta = self.ctx.store_meta.lock().unwrap();
-            meta.set_region(&self.ctx.coprocessor_host, cp.region, &mut self.fsm.peer);
+            meta.set_region(
+                &self.ctx.coprocessor_host,
+                cp.region.clone(),
```
Comment (hicqu) [inline src/raftstore/store/fsm/peer.rs:1470]: Is it required?

File: src/raftstore/store/fsm/peer.rs [hunk 2]
```diff
@@ -1466,6 +1489,7 @@ impl<'a, T: Transport, C: PdClient> PeerFsmDelegate<'a, T, C> {
                 }
                 self.fsm.peer.recent_conf_change_time = now;
                 self.fsm.peer.insert_peer_cache(peer);
+                // Update group config
```
Comment (hicqu) [inline src/raftstore/store/fsm/peer.rs:1492]: Do you mean we also need to know the peer's location?

File: src/raftstore/store/fsm/peer.rs (hunk 1 above)
Comment (Fullstop000) [reply src/raftstore/store/fsm/peer.rs:1470]: Otherwise, `cp` is moved by `set_region`.

File: src/raftstore/store/fsm/peer.rs (hunk 2 above)
Comment (Fullstop000) [reply src/raftstore/store/fsm/peer.rs:1492]: Sry, I forgot to remove this comment. The config will be updated in L1521

**Pull request merged**